# Numerical Data Scaling & Normalization

## Why Scale?

Also See: https://scikit-learn.org/stable/auto_examples/preprocessing/plot_scaling_importance.html#effect-of-rescaling-on-a-k-neighbors-models

In [3]:
import pandas as pd

In [7]:
house_price = 1_000

house_prices = [
    500_000,
    600_000,
    700_000,
    550_000,
    450_000,
    800_000,
    1_200_000,
    900_000,
]

In [8]:
jacket_price = 1_000
jacket_prices = [
    100,
    200,
    300,
    400,
    500,
    450,
    650,
    1200,
]

In [9]:
df = pd.DataFrame({
    'house_prices': house_prices,
    'jacket_prices': jacket_prices
})
df

house_prices  jacket_prices
0        500000            100
1        600000            200
2        700000            300
3        550000            400
4        450000            500
5        800000            450
6       1200000            650
7        900000           1200

In [11]:
df['house_price_z'] = (df['house_prices'] - df['house_prices'].mean()) / df['house_prices'].std()
df['jacket_price_z'] = (df['jacket_prices'] - df['jacket_prices'].mean()) / df['jacket_prices'].std()
df

house_prices  jacket_prices  house_price_z  jacket_price_z
0        500000            100      -0.853666       -1.102396
1        600000            200      -0.451941       -0.808424
2        700000            300      -0.050216       -0.514452
3        550000            400      -0.652804       -0.220479
4        450000            500      -1.054529        0.073493
5        800000            450       0.351510       -0.073493
6       1200000            650       1.958411        0.514452
7        900000           1200       0.753235        2.131300

In [18]:
df.loc[df['jacket_prices'] == 1200, ['jacket_prices', 'jacket_price_z']]

jacket_prices  jacket_price_z
7           1200          2.1313

In [17]:
df.loc[df['house_prices'] == 500000, ['house_prices', 'house_price_z']]

house_prices  house_price_z
0        500000      -0.853666

### Conclusion: Jacket vs House

Thus, we can say, based on the data we have, that a house price of 500,000 (z-score = -0.85) is much cheaper when placed on a fair scale compared to a jacket price of 1200 (z-score = 2.13). In fact, since the house price is negative, it is on the low-end of the scale.

## Why Noramlize?

See: https://scikit-learn.org/stable/auto_examples/compose/plot_transformed_target.html